src/utils/ValidatorUtils.ts

```diff
@@ -28,9 +29,38 @@ const DepositSchema = object({
   logIndex: Min(integer(), 0),
   quoteBlockNumber: Min(integer(), 0),
   transactionHash: HexValidator,
-  blockTimestamp: optional(Min(integer(), 0)),
+});
+
+const V3DepositSchema = object({
+  depositId: Min(integer(), 0),
+  depositor: AddressValidator,
+  recipient: AddressValidator,
+  inputToken: AddressValidator,
+  inputAmount: BigNumberValidator,
+  originChainId: Min(integer(), 0),
+  destinationChainId: Min(integer(), 0),
+  quoteTimestamp: Min(integer(), 0),
+  fillDeadline: Min(integer(), 0),
+  exclusivityDeadline: Min(integer(), 0),
+  exclusiveRelayer: AddressValidator,
+  realizedLpFeePct: optional(BigNumberValidator),
+  outputToken: AddressValidator,
+  outputAmount: BigNumberValidator,
+  message: string(),
+  speedUpSignature: optional(string()),
+  updatedOutputAmount: optional(BigNumberValidator),
+  updatedRecipient: optional(string()),
+  updatedMessage: optional(string()),
+  blockNumber: Min(integer(), 0),
+  transactionIndex: Min(integer(), 0),
+  logIndex: Min(integer(), 0),
+  quoteBlockNumber: Min(integer(), 0),
+  transactionHash: HexValidator,
 });
 
 export function isDepositFormedCorrectly(deposit: unknown): deposit is DepositWithBlock {
-  return DepositSchema.is(deposit);
+  return (
+    (V2DepositSchema.is(deposit) && isV2Deposit(deposit as DepositWithBlock)) ||
+    (V3DepositSchema.is(deposit) && isV3Deposit(deposit as DepositWithBlock))
```

Comment:
If we're validating the entire V2 or V3 schema, why do we need to call the isV2/V3Deposit type guards? Aren't those just a subset of the superstruct schema?

Reply:
The type assertion on input to `isV2Deposit()` should be unnecessary now, in any case.